### PR TITLE
Adding pull request and main push acceptance tests runner

### PR DESCRIPTION
**Pull request by mawasile**
This pull request includes several updates to Terraform configurations, GitHub workflows, and internal package structures. The changes enhance provider configurations, improve testing workflows, and refactor internal package naming for better clarity and maintainability.

### Terraform Configuration Updates:
* Added `azurecaf` and `azurerm` providers in `.devcontainer/features/acceptance_test_dependencies/main.tf` and `docs/resources/billing_policy_environment.md`. [[1]](diffhunk://#diff-4ff9366ad381ec4048a2dd7f3a067141eed94d774e620481a2baa0a14f6695e7R15-R20) [[2]](diffhunk://#diff-cebb69d0be225dbab2bdaa4c3b97c4231738b89c367861236a1e3e2fb90a674eR21-R83)
* Removed version constraints from `azurerm` and `azurecaf` providers in `docs/resources/billing_policy.md` and `examples/resources/powerplatform_billing_policy/resource.tf`. [[1]](diffhunk://#diff-47e72a573a0b8a0697c1b7bc0cd3e48cc283c14bec7dbd1f08bc9efc95ba114bL30-L38) [[2]](diffhunk://#diff-464317fcef3f894fcccbebbc851f072deaae69125379892c1cac8bae8292e638L5-L13)

### GitHub Workflow Enhancements:
* Added `pull_request` trigger and updated paths for `push` trigger in `.github/workflows/run_tests.yml`.
* Commented out the `unit_test` job and enhanced the `acceptance_test` job to include Terraform setup and local provider installation in `.github/workflows/run_tests.yml`. [[1]](diffhunk://#diff-c2e9f7c1bfaf65da79bbd29ed3ce389b2acc8b4099a257914a7292a66e6b9bc9L79-R114) [[2]](diffhunk://#diff-c2e9f7c1bfaf65da79bbd29ed3ce389b2acc8b4099a257914a7292a66e6b9bc9R142-R180)
* Updated `go test` commands to include `go clean -testcache` and `-failfast` options in `.github/workflows/terraform_provider.yml`. [[1]](diffhunk://#diff-9fa590c928d7324e2af24ae397276bc6a73b89ef6082dfb0eb440b17f527dc0fL134-R135) [[2]](diffhunk://#diff-9fa590c928d7324e2af24ae397276bc6a73b89ef6082dfb0eb440b17f527dc0fL171-R173)

### Internal Package Refactoring:
* Renamed internal packages from `powerplatform` to `api` in several files (`api_client.go`, `auth.go`, `lifecycle.go`, `request.go`). [[1]](diffhunk://#diff-03bd6d50cb06de9a7c49a1106113e5d0628b0c8b679d1816973a3e60c741c999L4-R21) [[2]](diffhunk://#diff-e16d073d63627064342ce1318299b7634406d3672eeba793dcbb1bceb7735aecL4-R4) [[3]](diffhunk://#diff-cf0f0685db3209a56763a1a0ee6440c06f7c401d8e47e1d4807fca6ea4ad00c5L4-R4) [[4]](diffhunk://#diff-01f4bec25997460df7ff85115dd42affd0671c0064322530f78a09891804bdd8L4-R4)
* Moved `constants` package to `internal/powerplatform/constants` and renamed it for better clarity. [[1]](diffhunk://#diff-5c2512dcf0c5474fe18d3cf401c768363775604d97e395a7c7caa9942d9d3a3fL4-R4) [[2]](diffhunk://#diff-5c2512dcf0c5474fe18d3cf401c768363775604d97e395a7c7caa9942d9d3a3fR15-R16)

### Error Handling and Logging Improvements:
* Enhanced error handling in `ExecuteForGivenScope` and `doRequest` methods in `internal/powerplatform/api/api_client.go` and `request.go`. [[1]](diffhunk://#diff-03bd6d50cb06de9a7c49a1106113e5d0628b0c8b679d1816973a3e60c741c999L61-R63) [[2]](diffhunk://#diff-01f4bec25997460df7ff85115dd42affd0671c0064322530f78a09891804bdd8L42-R56)
* Added retry logic with context timeout in `Execute` method in `internal/powerplatform/api/api_client.go`.

These changes collectively improve the maintainability, reliability, and clarity of the codebase, while also enhancing the CI/CD workflows to better handle Terraform provider dependencies and testing.

### CODE REVIEW

File: .github/workflows/run_tests.yml [hunk 1]
```diff
@@ -76,37 +79,37 @@ jobs:
...
-  unit_test:
-    name: Unit Test
-    needs: build
-    runs-on: ubuntu-latest
-    steps:
+  # unit_test:
+  #   name: Unit Test
+  #   needs: build
+  #   runs-on: ubuntu-latest
+  #   steps:
```
Comment (mawasile) [inline .github/workflows/run_tests.yml:87]: add concurency

File: internal/powerplatform/provider/provider.go [hunk 2]
```diff
@@ -517,3 +515,29 @@ func EnvDefaultFunc(k string, dv interface{}) string {
 	}
 	return ""
 }
+
+const (
+	// TestsUnitProviderConfig is a shared configuration to combine with the actual
+	// test configuration so the Power Platform client is properly configured.
+	// It is also possible to use the POWER_PLATFORM_ environment variables instead.
+	//lintignore:AT004
```
Comment (mattdot) [inline internal/powerplatform/provider/provider.go:523]: See if we can fix this linter warning instead of suppressing

**Review by mattdot — APPROVED** [review]

File: .github/workflows/run_tests.yml [hunk 3]
```diff
@@ -133,19 +138,44 @@ jobs:
...
+    - name: "Set up Terraform"
+      uses: hashicorp/setup-terraform@v3
+
+    - name: Download Terraform Providers
+      run: |
+          mkdir deps
+          cp .devcontainer/features/acceptance_test_dependencies/main.tf deps/main.tf
+          cd deps
+          terraform init
+          find .terraform/providers/ -type f -exec cp {} . \;
```
[the remark this answers is not in the document]
Comment (mawasile) [reply .github/workflows/run_tests.yml:151]: to investigate in a separate work item

File: internal/powerplatform/provider/provider.go (hunk 2 above)
Comment (mawasile) [reply internal/powerplatform/provider/provider.go:523]: to investigate in a separate work item

**Review by mattdot — APPROVED** [review]

**Review by leighatami — APPROVED** [review]

**Pull request merged**